app/build.gradle

```diff
@@ -1,64 +1,69 @@
+    compileSdk 35
+
+    namespace "com.github.vipulasri.timelineview.sample"
 
     defaultConfig {
         applicationId "com.github.vipulasri.timelineview.sample"
-        minSdkVersion versions.minSdk
-        targetSdkVersion versions.targetSdk
+        minSdk 21
+        targetSdk 35
         versionCode 5
         versionName "1.1"
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
         vectorDrawables.useSupportLibrary = true
     }
+
     buildTypes {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
-    androidExtensions {
-        experimental = true
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+
+    kotlinOptions {
+        jvmTarget = '17'
     }
-    viewBinding {
-        enabled = true
+
+    buildFeatures {
+        viewBinding true
     }
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
-    androidTestImplementation 'androidx.test:runner:1.2.0'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    androidTestImplementation 'androidx.test:runner:1.6.2'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.6.1'
+    implementation 'com.google.android.material:material:1.12.0'
 
     //kotlin
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${versions.kotlin.stdlib}"
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.10"
 
     //androidx
-    implementation "androidx.core:core-ktx:${versions.androidx.ktx.core}"
+    implementation 'androidx.core:core-ktx:1.13.1'
 
-    implementation "androidx.appcompat:appcompat:${versions.androidx.appCompat}"
-    implementation "androidx.constraintlayout:constraintlayout:${versions.androidx.constraintLayout}"
-    implementation "com.google.android.material:material:${versions.android.material}"
+    implementation "androidx.appcompat:appcompat:1.7.0"
+    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    implementation "com.google.android.material:material:1.12.0"
 
     //misc
-    implementation 'com.dmitrymalkovich.android:material-design-dimens:1.4'
+    implementation 'com.dmitrymalkovich.android:material-design-dimens:1.2'
     implementation ('com.thebluealliance:spectrum:0.7.1') {
         exclude group: 'com.android.support'
         exclude module: 'appcompat-v7'
         exclude module: 'support-v4'
     }
-    implementation ('org.adw.library:discrete-seekbar:1.0.1') {
-        exclude group: 'com.android.support'
-        exclude module: 'appcompat-v7'
-        exclude module: 'support-v4'
-    }
     implementation 'com.jakewharton.threetenabp:threetenabp:1.2.1'
     implementation project (':timelineview')
-    //implementation 'com.github.vipulasri:timelineview:1.1.0'
+    implementation project (':discreteseekbar')
```

Comment:
Why are we adding whole library project as dependency?

Reply:
Because discreteSeekBar is not working on your existing project.